src/Prefix.jl

```diff
@@ -213,6 +213,12 @@ function symlink_tree(src::AbstractString, dest::AbstractString)
             src_file = joinpath(root, f)
             dest_file = joinpath(dest, relpath(root, src), f)
             if isfile(dest_file)
+                src_file_hash = open(io -> bytes2hex(sha256(io)), src_file, "r")
```

Comment:
Can we have a quick `if filesize(dest_file) != filesize(src_file)` check?  That should prevent us from having to hash in the majority of cases, and should be much faster.

Reply:
Done!  And I've also added a test about this.